assets/shared/blocks/share-style.js

```diff
@@ -0,0 +1,65 @@
+import { BaseControl, Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { pick } from 'lodash';
+
+/**
+ * Add a control to apply color and style settings to all of the same blocks.
+ *
+ * @param {Object}   props
+ * @param {string}   props.clientId             Current block ID.
+ * @param {Object}   props.attributes           Current block attributes.
+ * @param {string[]} props.sharedAttributeNames Attributes to share between modules.
+ * @param {string}   props.help                 Help text.
+ * @param {string}   props.label                Button label text.
+ */
+export const ShareStyle = ( {
+	clientId,
+	attributes,
+	sharedAttributeNames,
+	help,
+	label,
+} ) => {
+	const getModuleBlocks = useSelect(
+		( select ) => {
+			return () => {
```

Comment:
Out of curiosity, are we using a function here because we're memoizing the `useSelect` function with the `clientId`?